### PR TITLE
add MAX_MONEY public constant to Amount

### PR DESCRIPTION
**Pull request by z8674558**
Closes https://github.com/rust-bitcoin/rust-bitcoin/issues/740

### CODE REVIEW

**Review by Kixunil — COMMENTED** [review]
Could you add the same constant for unsigned `Amount` too, please?

Note for other reviewers: `Mul::mul` is not `const` so `21_000_000 * Self::ONE_BTC` wouldn't work.

**Comment by z8674558** [comment]
@Kixunil thanks, updated.

**Comment by sanket1729** [comment]
We should address why @dpc reacted with thumbs down before we proceed on this.

**Review by Kixunil — CHANGES_REQUESTED** [review]

**Comment by Kixunil** [comment]
@sanket1729 I think he thought it is an implementation of #620 but it isn't. @dpc could you clarify?

File: src/util/amount.rs [hunk 1]
```diff
@@ -270,6 +270,8 @@ impl Amount {
     pub const ONE_SAT: Amount = Amount(1);
     /// Exactly one bitcoin.
     pub const ONE_BTC: Amount = Amount(100_000_000);
+    /// The maximum value allowed as an amount. Useful for sanity checking.
+    pub const MAX_MONEY: SignedAmount = SignedAmount(21_000_000 * 100_000_000);
```
[the remark this answers is not in the document]
Comment (z8674558) [reply src/util/amount.rs]: 🙏

**Comment by dpc** [comment]
Correct. Please don't mind me.

**Review by Kixunil — APPROVED** [review]
ACK ab12410ae87231d4bb8c0c2886f87cd2a2072aeb

**Review by RCasatta — APPROVED** [review]
ACK ab12410ae87231d4bb8c0c2886f87cd2a2072aeb

**Pull request merged**